lib/codecs/src/decoding/framing/chunked_gelf.rs

```diff
@@ -818,4 +979,290 @@ mod tests {
+        let mut decoder = ChunkedGelfDecoder {
+            decompression_config: ChunkedGelfDecompressionConfig::Gzip,
+            ..Default::default()
+        };
+
+        let error = decoder
+            .decode_eof(&mut compressed_payload.into())
+            .expect_err("decoding should fail");
+
+        let downcasted_error = downcast_framing_error(&error);
+        assert!(matches!(
+            downcasted_error,
+            ChunkedGelfDecoderError::Decompression {
+                source: ChunkedGelfDecompressionError::GzipDecompression { .. }
+            }
+        ));
+    }
+
+    #[tokio::test]
+    async fn decode_uncompressed_payload_with_gzip_decoder() {
+        let payload = "foo";
+        let mut decoder = ChunkedGelfDecoder {
+            decompression_config: ChunkedGelfDecompressionConfig::Gzip,
+            ..Default::default()
+        };
+
+        let error = decoder
+            .decode_eof(&mut payload.into())
+            .expect_err("decoding should fail");
+
+        let downcasted_error = downcast_framing_error(&error);
+        assert!(matches!(
+            downcasted_error,
+            ChunkedGelfDecoderError::Decompression {
+                source: ChunkedGelfDecompressionError::GzipDecompression { .. }
+            }
+        ));
+    }
+
+    #[tokio::test]
+    #[rstest]
+    #[case::gzip(Compression::Gzip)]
+    #[case::zlib(Compression::Zlib)]
+    async fn decode_compressed_payload_with_no_decompression_decoder(
+        #[case] compression: Compression,
+    ) {
+        let payload = "foo";
+        let compressed_payload = compression.compress(&payload);
+        let mut decoder = ChunkedGelfDecoder {
+            decompression_config: ChunkedGelfDecompressionConfig::None,
+            ..Default::default()
+        };
+
+        let frame = decoder
+            .decode_eof(&mut compressed_payload.clone().into())
+            .expect("decoding should not fail")
+            .expect("decoding should return a frame");
+
+        assert_eq!(frame, compressed_payload);
+    }
+
+    #[rstest]
+    #[case::level_zero(flate2::Compression::new(0))]
+    #[case::level_one(flate2::Compression::new(1))]
+    #[case::level_two(flate2::Compression::new(2))]
+    #[case::level_three(flate2::Compression::new(3))]
+    #[case::level_four(flate2::Compression::new(4))]
+    #[case::level_five(flate2::Compression::new(5))]
+    #[case::level_six(flate2::Compression::new(6))]
+    #[case::level_seven(flate2::Compression::new(7))]
+    #[case::level_eight(flate2::Compression::new(8))]
+    #[case::level_nine(flate2::Compression::new(9))]
+    fn detect_gzip_compression(#[case] level: flate2::Compression) {
+        let payload = "foo";
+        let compressed_payload = Compression::Gzip.compress_with_level(&payload, level);
+
+        let detected_compression = ChunkedGelfDecompression::from_magic(&compressed_payload);
+
+        assert_eq!(detected_compression, ChunkedGelfDecompression::Gzip);
+    }
```

Comment:
```suggestion
#[test]
fn detect_gzip_compression() {
    let payload = "foo";
    
    for level in 0..=9 {
        let level = flate2::Compression::new(level);
        let compressed_payload = Compression::Gzip.compress_with_level(&payload, level);

        let actual = ChunkedGelfDecompression::from_magic(&compressed_payload);

        assert_eq!(actual, ChunkedGelfDecompression::Gzip, "Failed for level {}", level);
    }
}
```

Reply:
Good catch! I overcomplicated it a bit :smile: 6f97a2ff2d0c4d11e2efc3e20620a101111b0a67